src/SlackConnector/SlackConnection.cs

```diff
@@ -63,20 +63,26 @@ public void Initialise(ConnectionInformation connectionInformation)
             ConnectedSince = DateTime.Now;
         }
 
-        private async Task ListenTo(InboundMessage inboundMessage)
+        private Task ListenTo(InboundMessage inboundMessage)
         {
-            if (inboundMessage?.MessageType != MessageType.Message)
-                return;
-            if (string.IsNullOrEmpty(inboundMessage.User))
-                return;
-            if (!string.IsNullOrEmpty(Self.Id) && inboundMessage.User == Self.Id)
-                return;
+            if (inboundMessage == null) Task.FromResult(false);
 
-            if (inboundMessage.Channel != null && !_connectedHubs.ContainsKey(inboundMessage.Channel))
+            switch (inboundMessage.MessageType)
             {
-                _connectedHubs[inboundMessage.Channel] = _chatHubInterpreter.FromId(inboundMessage.Channel);
+                case MessageType.Message: return HandleMessage((ChatMessage)inboundMessage);
+                case MessageType.Group_Joined: return HandleGroupJoined((GroupJoinedMessage)inboundMessage);
+                case MessageType.Channel_Joined: return HandleChannelJoined((ChannelJoinedMessage)inboundMessage);
             }
 
+            return Task.FromResult(false);
```

Comment:
Why is everything returning false for what is a void in async land? Could this be returned to using async instead?